hera_pspec/grouping.py

```diff
@@ -288,74 +308,88 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
                 # Iterate within a baseline-pair group and get integration-
                 # weighted data
                 for k, blp in enumerate(blpg):
-
                     # Get no. samples and construct integration weight
                     nsmp = uvp.get_nsamples((spw, blp, p))[:, None]
+                    # shape of nsmp: (Ntimes, 1)
                     data = uvp.get_data((spw, blp, p))
+                    # shape of data: (Ntimes, Ndlys)
                     wgts = uvp.get_wgts((spw, blp, p))
+                    # shape of wgts: (Ntimes, Ndlys, 2)
                     ints = uvp.get_integrations((spw, blp, p))[:, None]
-                    w = (ints * np.sqrt(nsmp))
+                    # shape of ints: (Ntimes, 1)
                     if store_cov:
                         cov = uvp.get_cov((spw, blp, p))
-                    # Get error bar weights and set invalid ones to zero.
+                        # shape of cov: (Ntimes, Ndlys, Ndlys)
+                    # Get error bar
                     errws = {}
                     for stat in stat_l:
-                        errs = uvp.get_stats(stat, (spw, blp, p)).copy()
-                        no_data = np.isnan(errs)
-                        errs[~no_data] = 1. / errs[~no_data] ** 2.
-                        errs[no_data] = 0.
-                        errws[stat] = errs
-
+                        errws[stat] = (uvp.get_stats(stat, (spw, blp, p)))**2
+                        # shape of errs: (Ntimes, Ndlys)
+
+                    if use_error_weights:
+                    # If use_error_weights==True, all arrays are weighted by a specified kind of errors,
+                    # including the error_filed in stats_array and cov_array. 
+                    # For each power spectrum P_i with error sigma_i, 
+                    # P_avg = \sum{ P_i / (sigma_i)^2 } / \sum{ 1 / (sigma_i)^2 } 
+                    # while for terms like variance or covariance,
+                    # epsilon_avg = \sum{ (epsilon_i / (sigma_i)^4 } / ( \sum{ 1 / (sigma_i)^2 } )^2
+                    # For reference: M. Tegmark 1997, The Astrophysical Journal Letters, 480, L87, Table 1, #3
+                    # or J. Dillon 2014, Physical Review D, 89, 023002 , Equation 34. 
+                        w = 1./(uvp.get_stats(error_weights, (spw, blp, p)))**2
+                        # shape of w: (Ntimes, Ndlys)
+                    else:
+                    # Otherwise all arrays are averaged in a way weighted by the integration time,
+                    # including the error_filed in stats_array and cov_array. 
+                        w = (ints * np.sqrt(nsmp))
```

Comment:
If we are inverse variance weighting I think this should be `w = (ints * np.sqrt(nsmp))**2`

Reply:
I agree.

Reply:
And nsmp here refers to N_coherent? So maybe we should have  w = (ints * nsmp)**2

Reply:
actually, Nsmp is Nincoherent, so it is correct the way it is I believe

Reply:
Yes, I have checked that and followed w = (ints * np.sqrt(nsmp))**2.